ServerCore/Pages/Teams/Details.cshtml.cs

```diff
@@ -163,13 +163,58 @@ public async Task<IActionResult> OnGetAddMemberAsync(int teamId, int userId, int
             {
                 return Forbid();
             }
-            
+
             Tuple<bool, string> result = TeamHelper.AddMemberAsync(_context, Event, EventRole, teamId, userId).Result;
             if (result.Item1)
             {
                 return RedirectToPage("./Details", new { teamId = teamId });
             }
             return NotFound(result.Item2);
         }
+
+        public async Task<IActionResult> OnGetRejectMemberAsync(int teamId, int userId, int applicationId)
+        {
+            var authResult = await AuthChecks(teamId);
+            if (!authResult.passed)
+            {
+                return authResult.redirect;
+            }
+
+            if (EventRole == EventRole.play && !Event.IsTeamMembershipChangeActive)
+            {
+                return NotFound("Team membership change is not currently active.");
+            }
+
+            TeamApplication application = await (from app in _context.TeamApplications
+                                                 where app.ID == applicationId
+                                                 select app).FirstOrDefaultAsync();
+            if (application == null)
+            {
+                return NotFound("Could not find application");
+            }
+
+            if (application.Player.ID != userId)
+            {
+                return NotFound("Mismatched player and application");
+            }
+            Team team = await _context.Teams.FirstOrDefaultAsync(m => m.ID == teamId);
+            if (application.Team != team)
+            {
+                return Forbid();
+            }
+
+            var allApplications = from app in _context.TeamApplications
```

Comment:
...in what case can this be a list instead of one thing?

Reply:
All the other cases treated it as a list so I did too. I don't think it can be more than one thing either.